platform/core/src/classes/HotkeysManager.js

```diff
@@ -48,18 +48,34 @@ export class HotkeysManager {
    * default hotkey bindings for all provided definitions. These
    * values are used in `this.restoreDefaultBindings`.
    *
-   * @param {HotkeyDefinition[]} hotkeyDefinitions
+   * @param {HotkeyDefinition[] | Object} hotkeyDefinitions
    * @param {Boolean} [isDefaultDefinitions]
    */
   setHotkeys(hotkeyDefinitions, isDefaultDefinitions = false) {
-    const definitions = cloneDeep(hotkeyDefinitions);
+    const definitions = Array.isArray(hotkeyDefinitions)
+      ? [...hotkeyDefinitions]
+      : this.parseToArrayLike(hotkeyDefinitions);
     definitions.forEach(definition => this.registerHotkeys(definition));
 
     if (isDefaultDefinitions) {
       this.hotkeyDefaults = definitions;
     }
   }
 
+  parseToArrayLike(hotkeyDefinitionsObj) {
```

Comment:
What about jsDoc here? also, let's rename to _parseToArrayLike to keep the consistency of our private methods naming convention.